src/init.cpp

```diff
@@ -1916,6 +1920,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         if (index == std::string::npos) {
             bind_addr = Lookup(bind_arg, default_bind_port, /*fAllowLookup=*/false);
             if (bind_addr.has_value()) {
+                connOptions.bind_on_any |= bind_addr.value().IsBindAny();
```

Comment:
If the semantic of `bind_on_any` is to be changed then its comment deserves an update:

https://github.com/bitcoin/bitcoin/blob/5b8046a6e893b7fad5a93631e6d1e70db31878af/src/net.h#L1072-L1074